parser/openfoodfacts_taxonomy_parser/parser/taxonomy_parser.py

```diff
@@ -464,6 +466,10 @@ def _merge_duplicate_entry_nodes(self, entry_nodes: list[NodeData]) -> list[Node
         for node in entry_nodes:
             if node.id in ids_to_nodes:
                 first_node = ids_to_nodes[node.id]
+                # if we merge a node from the taxonomy with an external node,
+                # the merged node is not external anymore
+                if first_node.is_external and not node.is_external:
+                    first_node.is_external = False
```

Comment:
:+1: great you though of this.

That said, I'm not sure we want to support it !

I would even think about doing the reverse: even if a node is present in the taxonomy but is also in an external one, it should not be edited.
Otherwise, on export we will silently copy information from the other taxonomy and create duplication silently. 

On the other hands, doing as I just propose would mean removing an entry from the taxonomy :thinking: 

So I'm not sure what to do…

I would rather not support it yet… could we imagine keeping node separate in this case ? As it is a potential problem to have twice same id, I would change the external id to <id>@<external taxonomy name> (eg: en:cochinine@additives.txt) and leave it like that for now…